src/main/js/dashboard-frontend/src/components/PubSubDebugger.tsx

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, {useCallback, useContext, useRef, useState} from "react";
+import {withRouter} from "react-router-dom";
+
+import {
+    Box,
+    Button, CollectionPreferences, CollectionPreferencesProps,
+    Container,
+    ContentLayout,
+    Form,
+    FormField,
+    Grid,
+    Header,
+    Icon,
+    Input,
+    Link, Pagination, Select, SelectProps,
+    SpaceBetween,
+    Table,
+    Tabs, TabsProps,
+    Textarea,
+} from "@cloudscape-design/components";
+
+import {DefaultContext, SERVER} from "../index";
+import {APICall} from "../util/CommUtils";
+import {CommunicationMessage} from "../util/CommunicationMessage";
+import {useCollection} from "@cloudscape-design/collection-hooks";
+
+interface Message {
+    binaryPayload: string;
+    received: Date;
+    topic: string;
+}
+
+const PubSub = () => {
+    const [selectedTopic, setSelectedTopic] = useState("");
+    const [subscribeTopicInputValue, setSubscribeTopicInputValue] = useState("");
```

Comment:
This component might benefit of using `useReducer` to manage its state. Again, not something that is necessary but I find it to be helpful once you have enough useStates that are related to one another https://kentcdodds.com/blog/should-i-usestate-or-usereducer